blazor/dropdown-list/data-binding.md

```diff
@@ -7,68 +7,377 @@ control: DropDown List
+    /// Performs data Read operation asynchronously.
+    /// </summary>
+    public virtual Task<object> ReadAsync(DataManagerRequest dataManagerRequest, string key = null)
 }
 ```
 
-## Index Value Binding
+The custom data binding can be performed in the DropDownList component by providing the custom adaptor class and overriding the Read or ReadAsync method of the DataAdaptor abstract class.
+
+The following sample code demonstrates implementing custom data binding using custom adaptor,
+
+{% highlight cshtml %}
+
+{% include_relative code-snippet/data-binding/custom-adaptor.razor %}
+
+{% endhighlight %}
+
+### Offline mode
+
+To avoid post back for every action, set the DropDownList to load all data on initialization and make the actions process in client-side. To enable this behaviour, use the `Offline` property of `DataManager`.
+
+The following example for remote data binding and enabled offline mode.
+
+{% highlight cshtml %}
+
+{% include_relative code-snippet/data-binding/offline-mode.razor %}
+
+{% endhighlight %}
+
+![Blazor DropdownList with Offline mode](./images/data-binding/blazor_dropdown_offline-mode.png)
+
+### Entity framework
 
-Index value binding can be achieved by using `bind-Index` attribute and it supports int and int nullable types. By using this attribute you can bind the values respective to its index.
+You need to follow the below steps to consume data from the [Entity Framework](https://blazor.syncfusion.com/documentation/common/data-binding/bind-entity-framework) in the DropDownList component.
 
-```cshtml
-@using Syncfusion.Blazor.DropDowns
+#### Create DBContext class
 
-<SfAutoComplete TValue="string" Placeholder="e.g. Australia" TItem="Country" @bind-Index="@ddlIndex" DataSource="@Countries">
-    <AutoCompleteFieldSettings Value="Name"></AutoCompleteFieldSettings>
-</SfAutoComplete>
+The first step is to create a DBContext class called `OrderContext` to connect to a Microsoft SQL Server database.
 
-@code {
+```csharp
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using EFDropDown.Shared.Models;
 
-    private int? ddlIndex { get; set; } = 1;
+namespace EFDropDown.Shared.DataAccess
+{
+    public class OrderContext : DbContext
+    {
+        public virtual DbSet<Shared.Models.Order> Orders { get; set; }
+
+        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+        {
+            if (!optionsBuilder.IsConfigured)
+            {
+                optionsBuilder.UseSqlServer(@"Data Source=(LocalDB)\MSSQLLocalDB;AttachDbFilename=D:\Blazor\DropDownList\EFDropDown\Shared\App_Data\NORTHWND.MDF;Integrated Security=True;Connect Timeout=30");
+            }
+        }
+    }
+}
+```
+
+#### Create data access layer to perform data operation
+
+Now, you need to create a class named `OrderDataAccessLayer`, which act as data access layer for retrieving the records from the database table.
```

Comment:
Now, create a class named `OrderDataAccessLayer,` which acts as a data access layer for retrieving the records from the database table.